org.scala-refactoring.library/src/main/scala/scala/tools/refactoring/implementations/Rename.scala

```diff
@@ -79,8 +80,20 @@ abstract class Rename extends MultiStageRefactoring with TreeAnalysis with analy
 
     val rename = topdown(isInTheIndex &> renameTree |> id)
 
-    val renamedTrees = occurences flatMap (rename(_))
+    val renamedTreesWithOriginals = occurences.flatMap { tree =>
+      rename(tree).map((_, tree))
+    }
+
+    val renameSourceChanges = renamedTreesWithOriginals.flatMap {
+      case (newTree: ImplDef, oldTree: ImplDef) if newTree.name != oldTree.name && newTree.name != null && oldTree.name != null =>
+        if (oldTree.pos.isDefined && oldTree.pos.source.file.name == oldTree.name.toString + ".scala") {
+          Some(RenameSourceFileChange(oldTree.pos.source.file, newTree.name.toString() + ".scala"))
+        } else {
+          None
+        }
+      case _ => None
+    }
```

Comment:
Wouldn't `collect` be more readable here? It might be necessary to extract the if-check, but maybe that would also make it a bit more readable.